src/charm.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""A Juju charm for OpenTelemetry Collector on Kubernetes."""
+
+import os
+from typing import Any, Dict
+
+from config import ConfigManager
+
+from ops import CharmBase, main
+from ops.model import ActiveStatus
+from ops.pebble import Layer
+
+
+class OpenTelemetryCollectorK8sCharm(CharmBase):
+    """Charm to run OpenTelemetry Collector on Kubernetes."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        if not self.unit.get_container("opentelemetry-collector").can_connect():
```

Comment:
The `can_connect` guard is not reliable, look:

https://github.com/canonical/loki-k8s-operator/issues/434

We should use `tenacity` or a simple `try...except` block

Reply:
How is this different from having the `can_connect` guard fail?

The outcome we want is for Juju to retry:
- with `can_connect`, if it fails the charm will go into error state, and retry;
- with `try`/`except`, would we `raise`? If so, the charm will still go into error state;
- with `tenacity`, we're retrying internally instead of letting Juju do it, the only difference being the charm never goes into error state.

Leaving this up to discuss with the wider team